File: server/ruby/app/infrastructure/mappers/payment_details/hosted_checkout_specific_output_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class HostedCheckoutSpecificOutputMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          output = Business::Domain::Payments::PaymentDetails::HostedCheckoutSpecificOutput.new
          output.hosted_checkout_id = response.hosted_checkout_id
          output.variant = response.variant

          output
        end
      end
    end
  end
end

File: server/python/app/infrastructure/mappers/payment_details/hosted_checkout_specific_output_mapper.py
from typing import Optional
from app.application.domain.payments.payment_details.hosted_checkout_specific_output import HostedCheckoutSpecificOutput
from onlinepayments.sdk.domain.hosted_checkout_specific_output import HostedCheckoutSpecificOutput as HostedCheckoutSpecificOutputSdk

class HostedCheckoutSpecificOutputMapper:

    @staticmethod
    def map_from_sdk_response(response: Optional[HostedCheckoutSpecificOutputSdk]) -> Optional[HostedCheckoutSpecificOutput]:
        if response is None:
            return None

        output = HostedCheckoutSpecificOutput()
        output.hosted_checkout_id = response.hosted_checkout_id
        output.variant = response.variant

        return output